pyzipcode/__init__.py

```diff
@@ -64,21 +65,29 @@ def query(self, sql, params=None):
     "  AND latitude <= ?"
 )
 ZIP_FIND_QUERY = "SELECT * FROM ZipCodes WHERE city LIKE ? AND state LIKE ?"
+ZIP_ALL_QUERY = "SELECT zip FROM ZipCodes ORDER BY zip ASC"
+ZIP_ALL_REVERSED_QUERY = "SELECT zip FROM ZipCodes ORDER BY zip DESC"
+ZIP_LEN_QUERY = "SELECT COUNT(*) FROM ZipCodes"
 
 
-class ZipCode(object):
+class ZipCode:
     """
     Represents one zipcode record from the database.
     """
 
-    def __init__(self, data):
-        self.zip = data[0]
-        self.city = data[1]
-        self.state = data[2]
-        self.longitude = data[3]
-        self.latitude = data[4]
-        self.timezone = data[5]
-        self.dst = data[6]
+    def __init__(self, zip, city, state, longitude, latitude, timezone, dst):
```

Comment:
this is a backwards incompatible change. This could be done somewhat backwards-compatibly by making all but the first argument optional and also checking the type of the first argument.

Reply:
Much better.